src/Rules.Framework/Generics/GenericRulesEngine.cs

```diff
@@ -0,0 +1,48 @@
+namespace Rules.Framework.Generics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Rules.Framework.Extensions;
+
+    internal class GenericRulesEngine<TContentType, TConditionType> : IGenericRulesEngine
```

Comment:
Mark the class as sealed to increase its implementation performance. To understand why:

> When calling virtual methods, the actual method is found at runtime based on the actual type of the object. Each type has a Virtual Method Table ([vtable](https://en.wikipedia.org/wiki/Virtual_method_table)) which contains the address of all the virtual methods. These pointers are used at runtime to invoke the appropriate method implementations ([dynamic dispatch](https://en.wikipedia.org/wiki/Dynamic_dispatch)).
>
> If the JIT knows the actual type of the object, it can skip the vtable and call the right method directly to improve performance. Using sealed types helps the JIT as it knows there cannot be any derived class.

More details here: https://www.meziantou.net/performance-benefits-of-sealed-class.htm

Reply:
Done